src/components/application_manager/src/resumption/resume_ctrl_impl.cc

```diff
@@ -789,11 +966,35 @@ bool ResumeCtrlImpl::CheckIcons(ApplicationSharedPtr application,
   return mobile_apis::Result::INVALID_DATA != verify_images;
 }
 
-bool ResumeCtrlImpl::CheckDelayAfterIgnOn() {
+bool ResumeCtrlImpl::CheckIgnCyclesData() const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const uint32_t global_ign_off_count =
+      resumption_storage_->GetGlobalIgnOffCounter();
+  const uint32_t global_ign_on_count =
+      resumption_storage_->GetGlobalIgnOnCounter();
+  const uint32_t diff = global_ign_on_count - global_ign_off_count;
+  const bool is_ign_off_record_missed = diff >= 2;
+  if (is_ign_off_record_missed) {
+    LOG4CXX_WARN(logger_,
+                 "Some IGN OFF records missed. Possibly due to Low Voltage");
+    return false;
```

Comment:
@yang1070 I will reset global counters after normal IGNITION OFF flow

Reply:
@yang1070 We found a betters and more simple solution. We will use only one IGN_ON counter and will reset it in normal IGNITION OFF flow. Code will be provided in separate commit later.